src/main/java/com/checkmarx/flow/service/AbstractASTScanner.java

```diff
@@ -30,12 +36,21 @@
     private final FlowProperties flowProperties;
     private final String scanType;
     private final BugTrackerEventTrigger bugTrackerEventTrigger;
+    private final ScaProperties scaProperties;
+    private final CxRepoFileService cxRepoFileService;
 
     @Override
     public ScanResults scan(ScanRequest scanRequest) {
         ScanResults result = null;
         log.info("--------------------- Initiating new {} scan ---------------------", scanType);
         ScanParams sdkScanParams = toSdkScanParams(scanRequest);
+
+        if (scaProperties.isEnabledZipScan()) {
```

Comment:
I see SCA-specific logic inside AbstractASTScanner. Shouldn't this logic be in SCAScanner?

Reply:
AbstractASTScanner  holds the logic for the scan flow for both ast & sca.
Moreover, the SCAScanner extends the AbstractASTScanner and it's won't be a good practice to access a son class from parent.